### ISSUE TITLE
MethodCountCheck: class counts include anonymous methods

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#4539]
Identified in regression at https://github.com/checkstyle/checkstyle/pull/4535 :

````
$ cat TestClass.java
public class TestClass {
    void method() {
        Runnable r = (new Runnable() {
                public void run() {
                }
            });
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="MethodCountCheck">
  <property name="maxTotal" value="1" />
</module>
    </module>
</module>

$ java -jar checkstyle-7.8.2-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:1: Total number of methods is 2 (max allowed is 1). [MethodCount]
Audit done.
Checkstyle ends with 1 errors.
````

Method count is set to allow a max of 1 method per class.
Checkstyle incorrectly reports that the class `TestClass` has 2 methods. It is incorrectly including the anonymous method in the counting of `TestClass`. I don't think it should be included, as it is a separate class from `TestClass`.
I am expecting no violations on this code.

**Attention**: after fix, counts are expected to be lowered, so we should change our checkstyle_checks.xml

**Comment by rnveach** [comment]
> after fix, counts are expected to be lowered, so we should change our checkstyle_checks.xml

With fix, the lowest we can go without violations (or suppressions) is 34. Current value is 35.
Here are violations with 28:
````
Starting audit...
[ERROR] \checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\Checker.java:66: Total number of methods is 32 (max allowed is 30). [MethodCount]
[ERROR] \checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\checks\coding\FinalLocalVariableCheck.java:77: Total number of methods is 34 (max allowed is 30). [MethodCount]
[ERROR] \checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\checks\indentation\AbstractExpressionHandler.java:33: Total number of methods is 31 (max allowed is 30). [MethodCount]
[ERROR] \checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\TreeWalker.java:62: Total number of methods is 29 (max allowed is 25). [MethodCount]
Audit done.
Checkstyle ends with 4 errors.

````

**Comment by romani** [comment]
fix is merged

**Comment by rnveach** [comment]
@romani You made mention in this issue you wanted to change our numbers for the MethodCountCheck configuration. You no longer want to do this?

**Comment by romani** [comment]
execute:
     [echo] Checkstyle started (checkstyle_checks.xml): 01/08/2017 04:01:13 PM
> You no longer want to do this?

thanks a lot for pinning me, I missed this.
In my local:
```
[checkstyle] Running Checkstyle 8.2-SNAPSHOT on 938 files
[checkstyle] [ERROR] .../checks/coding/FinalLocalVariableCheck.java:77: 
Total number of methods is 34 (max allowed is 32). [MethodCount]
```

fixed in  https://github.com/checkstyle/checkstyle/commit/29e131f949950d26e7880da5891d1754776fdeee
